terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/cache.ts

```diff
@@ -278,36 +278,77 @@ export async function redisLocked<T>(
   throw new Error(`Could not acquire lock for ${nameSpace}-${key}`);
 }
 
-export async function getExperimentValue(experimentKey: string, defaultValue: number): Promise<number> {
-  await startupRedisPool();
-  if (!redisPool) throw Error('Redis should be initialized!');
+export async function getExperimentValueStr(experimentKey: string, defaultValue: string): Promise<string> {
+  return locallyCached('EXPERIMENT', experimentKey, 60, async (): Promise<string> => {
+    await startupRedisPool();
+    if (!redisPool) throw Error('Redis should be initialized!');
 
-  const queryKey = `${Config.Instance.environment}.EXPERIMENT.${experimentKey}`;
+    const queryKey = `${Config.Instance.environment}.EXPERIMENT.${experimentKey}`;
 
-  console.debug(`Checking for experiment ${experimentKey} (${queryKey})`);
-  try {
-    const experimentValue: string | undefined | null = await redisPool.use(async (client: RedisClientType) => {
-      return await client.get(queryKey);
-    });
-
-    if (experimentValue !== undefined && experimentValue !== null) {
-      const numValue = Number(experimentValue);
-      if (!isNaN(numValue)) {
-        console.debug(`Found experiment ${queryKey} with value ${numValue}`);
-        return numValue;
+    console.debug(`Checking redis entry for experiment ${experimentKey} (${queryKey})`);
+    try {
+      const experimentValue: string | undefined | null = await redisPool.use(async (client: RedisClientType) => {
+        return await client.get(queryKey);
+      });
+
+      if (experimentValue !== undefined && experimentValue !== null) {
+        return experimentValue;
       } else {
-        console.warn(`Experiment ${queryKey} found but value is not a valid number: ${experimentValue}`);
+        console.debug(`Experiment ${queryKey} not found, using default value ${defaultValue}`);
       }
+    } catch (e) {
+      console.error(`Error retrieving experiment ${queryKey}, using default value ${defaultValue}: ${e}`);
+    }
+
+    return defaultValue;
+  });
+}
+
+export async function getExperimentValue(experimentKey: string, defaultValue: number): Promise<number> {
+  try {
+    const experimentValue = await getExperimentValueStr(experimentKey, defaultValue.toString());
+    const numValue = Number(experimentValue);
+    if (!isNaN(numValue)) {
+      console.debug(`Found experiment ${experimentKey} with value ${numValue}`);
+      return numValue;
     } else {
-      console.debug(`Experiment ${queryKey} not found, using default value ${defaultValue}`);
+      console.warn(
+        `Experiment ${experimentKey} found but value is not a valid number:` +
+          ` ${experimentValue} - returning default value ${defaultValue}`,
+      );
     }
   } catch (e) {
-    console.error(`Error retrieving experiment ${queryKey}: ${e}`);
+    /* istanbul ignore next */
+    console.error(`Error retrieving experiment ${experimentKey} - returning default value ${defaultValue}: ${e}`);
```

Comment:
Is this catch block even possible to hit, given the try/catch inside getExperimentalValueStr?